item-counter-spe/src/index.tsx

```diff
@@ -0,0 +1,192 @@
+	// Set the active account
+	msalInstance.setActiveAccount(account);
+
+	// Create the GraphHelper instance
+	// This is used to interact with the Graph API
+	// Which allows the app to get the file storage container id, the Fluid container id,
+	// and the site URL.
+	const graphHelper = new GraphHelper(msalInstance, account);
+
+	// Define a function to get the container info based on the URL hash
+	// The URL hash is the shared item id and will be used to get the file storage container id
+	// and the Fluid container id. If there is no hash, then the app will create a new Fluid container
+	// in a later step.
+	const getContainerInfo = async () => {
+		const shareId = location.hash.substring(1);
+		if (shareId.length > 0) {
+			try {
+				return await graphHelper.getSharedItem(shareId);
+			} catch (error) {
+				showErrorMessage("Error while fetching shared item: ", error as string);
+				return undefined;
+			}
+		} else {
+			return undefined;
+		}
+	};
+
+	// Get the file storage container id (driveId) and the Fluid container id (itemId).
+	const containerInfo = await getContainerInfo();
+
+	// Define a function to get the file storage container id using the Graph API
+	// If the user doesn't have access to the file storage container, then the app will fail here.
+	const getFileStorageContainerId = async () => {
+		try {
+			return await graphHelper.getFileStorageContainerId();
+		} catch (error) {
+			showErrorMessage("Error while fetching file storage container ID: ", error as string);
+			return "";
+		}
+	};
+
+	let fileStorageContainerId = "";
+	let containerId = "";
+
+	// If containerInfo is undefined, then get the file storage container id using the function
+	// defined above.
+	// If the containerInfo is not undefined, then use the file storage container id and Fluid container id
+	// from containerInfo.
+	if (containerInfo === undefined) {
+		fileStorageContainerId = await getFileStorageContainerId();
+	} else {
+		fileStorageContainerId = containerInfo.driveId;
+		containerId = containerInfo.itemId;
+	}
+
+	// If the file storage container id is empty, then the app will fail here.
+	if (fileStorageContainerId.length == 0) {
+		return;
+	}
+
+	// Create the client properties required to initialize
+	// the Fluid client instance. The Fluid client instance is used to
+	// interact with the Fluid service.
+	const clientProps = getClientProps(
+		await graphHelper.getSiteUrl(),
+		fileStorageContainerId,
+		new SampleOdspTokenProvider(msalInstance),
+	);
+
+	// Create the Fluid client instance
+	const client = new OdspClient(clientProps);
+
+	// Create the root element for React
+	const app = document.createElement("div");
+	app.id = "app";
+	document.body.appendChild(app);
+	const root = createRoot(app);
+
+	// Initialize Fluid Container - this will either make a new container or load an existing one
+	const { container } = await loadFluidData(containerId, containerSchema, client);
```

Comment:
loadFluidData doesn't do too much - maybe just inline it here so readers have fewer jumps in their readthrough
```suggestion
	const { container } = containerId.length === 0 ? await client.createContainer(containerSchema) : await client.getContainer(containerSchema);
```